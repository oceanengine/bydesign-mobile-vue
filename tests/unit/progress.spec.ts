import { mount, later } from '../index';
import { Progress as NativeProgress } from '@src'
const Progress = NativeProgress as any;
describe('Progress', () => {
  test('textInMiddle props', async () => {
    const wrapper = mount(Progress, {
      propsData: {
        textInMiddle: true,
        percentage: 100,
      },
    });
    await later();
    expect(wrapper).toMatchSnapshot();
  });
  test('track color prop', async () => {
    const wrapper = mount(Progress, {
      propsData: {
        trackColor: 'green',
      },
    });
    await later();
    expect(wrapper.find('.byted-progress__portion').element.style.background).toEqual('green');
  });
});
