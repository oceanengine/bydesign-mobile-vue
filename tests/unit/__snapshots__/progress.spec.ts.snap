// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress textInMiddle props 1`] = `
<div class="byted-progress byted-progress--text-in-middle">
  <div class="byted-progress__portion" style="height: 4px; border-radius: 2px;">
    <div class="byted-progress__portioncon" style="width: 100%; border-radius: 2px;">
      <div class="byted-progress__pivot-middle">100%</div>
    </div>
  </div>
</div>
`;
